specs/AreBrowserAcceleratorKeysEnabled.md

```diff
@@ -0,0 +1,109 @@
+that correspond to browser-specific functionality, without disabling things like
+copy and paste and movement.
+
+In this document we describe the new setting. We'd appreciate your feedback.
+
+
+## Description
+When this setting is set to false, it disables all accelerator keys that access
+features specific to the browser, including but not limited to:
+- Ctrl-F and F3 for Find on Page
+- Ctrl-P for Print
+- Ctrl-R and F5 for Reload
+- Ctrl-Plus and Ctrl-Minus for zooming
+- Ctrl-Shift-C and F12 for DevTools
+- Special keys for browser functions, such as Back, Forward, and Search
+
+It does not disable accelerator keys related to movement and text editing, such
+as:
+- Home, End, Page Up, and Page Down
+- Ctrl-X, Ctrl-C, Ctrl-V
+- Ctrl-A for Select All
+- Ctrl-Z for Undo
+
+Those accelerator keys will always be enabled unless they are handled in the
+`AcceleratorKeyPressed` event.
+
+The default value for AreBrowserAcceleratorKeysEnabled is true.
+
+## Examples
+```c#
+void ToggleBrowserKeysEnabled()
+{
+    var settings = _webView.CoreWebView2.Settings;
+    settings.AreBrowserAcceleratorKeysEnabled = !settings.AreBrowserAcceleratorKeysEnabled;
+}
+```
+
+```cpp
+void SettingsComponent::ToggleBrowserKeysEnabled()
+{
+    BOOL enabled;
+    CHECK_FAILURE(_coreWebView2Settings->get_AreBrowserAcceleratorKeysEnabled(&enabled));
+    CHECK_FAILURE(_coreWebView2Settings->put_AreBrowserAcceleratorKeysEnabled(enabled ? FALSE : TRUE));
+}
+```
+
+
+## Remarks
+Some accelerator keys that don't make sense for WebViews are always disabled
+regardless of this setting.  This includes things like opening and closing tabs,
+viewing bookmarks and history, and selecting the location bar.
+
+This setting will not prevent commands from being run through the context menu.
+To disable the context menu, use `AreDefaultContextMenusEnabled`.
+
+
+## API Details
+```
+[uuid(9aab8652-d89f-408d-8b2c-1ade3ab51d6d), object, pointer_default(unique)]
+interface ICoreWebView2Settings2 : ICoreWebView2Settings {
+    /// When this setting is set to false, it disables all accelerator keys
+    /// that access features specific to the browser, including but not limited to:
+    ///  - Ctrl-F and F3 for Find on Page
+    ///  - Ctrl-P for Print
+    ///  - Ctrl-R and F5 for Reload
+    ///  - Ctrl-Plus and Ctrl-Minus for zooming
+    ///  - Ctrl-Shift-C and F12 for DevTools
+    ///  - Special keys for browser functions, such as Back, Forward, and Search
+    ///
+    /// It does not disable accelerator keys related to movement and text editing,
+    /// such as:
+    ///  - Home, End, Page Up, and Page Down
+    ///  - Ctrl-X, Ctrl-C, Ctrl-V
+    ///  - Ctrl-A for Select All
+    ///  - Ctrl-Z for Undo
+    ///
+    /// Those accelerator keys will always be enabled unless they are handled in the
+    /// `AcceleratorKeyPressed` event.
+    /// The default value for AreBrowserAcceleratorKeysEnabled is true.
```

Comment:
Is there a way to hear those events be raised and react to them, or intercept them as they bubble? Or is it all-or-nothing?

Reply:
AcceleratorKeyPressed is an event on CoreWebView2 that allows you to react to individual keys as you see fit, so if I understand the question correctly, yes.

Reply:
Lewis, please explicitly describe how the AcceleratorKeyPressed event works with AreBrowserAcceleratorKeysEnabled in the documentation. If you set AreBrowserAcceleratorKeysEnabled to false, will the event still raise for the disabled accelerator keys?

Reply:
So even if I set the new bool false, the event is still raised? (That would be a good thing.)

Reply:
Yes, the event will fire regardless of this setting.  We'll put that in the documentation.